test/openssl/utils.rb

```diff
@@ -106,6 +106,26 @@ def get_subject_key_id(cert, hex: true)
     end
   end
 
+  def generate_ed25519
+    # Ed25519 is not FIPS-approved.
+    omit_on_fips
+
+    begin
+      ed25519  = OpenSSL::PKey::generate_key("ED25519")
+    rescue OpenSSL::PKey::PKeyError => e
+      # OpenSSL < 1.1.1
+      #
+      pend "Ed25519 is not implemented" unless openssl?(1, 1, 1)
+
+      raise e
+    end
+
+    # See ASN1_item_sign_ctx in ChangeLog for 3.8.1: https://github.com/libressl/portable/blob/master/ChangeLog
+    pend 'ASN1 signing with Ed25519 not yet working' unless openssl? or libressl?(3, 8, 1)
+
+    ed25519
```

Comment:
`generate_ed25519` is vague about what it is for. LibreSSL 3.8.1 is the version that supported Ed25519 through `ASN1_item_sign()` (and hence `X509_*_sign()`). `OpenSSL::PKey#sign`, for example, supports Ed25519 since an earlier version. Also, `omit` in a helper method like this is unexpected for me.

That said, the condition can be greatly simplified. I think it's acceptable to repeat this a few times in our test code.

```suggestion
    omit_on_fips
    omit "Ed25519 not supported" unless openssl?(1, 1, 1) or libressl?(3, 8, 1)
    OpenSSL::PKey::generate_key("ED25519")
```